From 6aca28bc7ece1f3dcb1e1c90246a05705aecb265 Mon Sep 17 00:00:00 2001
From: Martine Lenders <m.lenders@fu-berlin.de>
Date: Thu, 31 Mar 2022 12:08:03 +0200
Subject: [PATCH] nanocoap: expose coap_find_option()

This function was originally exposed in PR17474 in master, which was
merged after 2022.01 was released. It is needed however for
coap_opt_remove() introduced in PR17881 and required by client caching
(introduced in PR17888).
---
 sys/include/net/nanocoap.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/sys/include/net/nanocoap.h b/sys/include/net/nanocoap.h
index 25c27ea969..600aebcc35 100644
--- a/sys/include/net/nanocoap.h
+++ b/sys/include/net/nanocoap.h
@@ -435,6 +435,17 @@ static inline void coap_hdr_set_type(coap_hdr_t *hdr, unsigned type)
  * Read options from a parsed packet.
  */
 /**@{*/
+/**
+ * @brief   Get pointer to an option field by type
+ *
+ * @param[in]   pkt     packet to work on
+ * @param[in]   opt_num the option number to search for
+ *
+ * @returns     pointer to the option data
+ *              NULL if option number was not found
+ */
+uint8_t *coap_find_option(const coap_pkt_t *pkt, unsigned opt_num);
+
 /**
  * @brief   Get content type from packet
  *
-- 
2.35.1

